src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -1014,7 +1018,11 @@ private async Task DoMaintenanceAsync()
             foreach (var kvp in _memory.ToArray())
             {
                 bool lastAccessTimeIsInfrequent = kvp.Value.LastAccessTicks < lastAccessMaximumTicks;
-                if (lastAccessTimeIsInfrequent && kvp.Value.ExpiresAt < DateTime.MaxValue && kvp.Value.ExpiresAt <= utcNow)
+                if (!lastAccessTimeIsInfrequent)
+                    continue;
+
+                var expiresAt = kvp.Value.ExpiresAt;
+                if (expiresAt < DateTime.MaxValue && expiresAt <= utcNow)
```

Comment:
![image](https://github.com/user-attachments/assets/48351b98-9164-4cf9-8fa6-1868fb9431d0)

Just trying to cut down a tiny bit on number of times expires is called.